src/me/StevenLawson/TotalFreedomMod/Commands/Command_trail.java

```diff
@@ -18,16 +18,42 @@
 import org.bukkit.plugin.RegisteredListener;
 
 @CommandPermissions(level = AdminLevel.SUPER, source = SourceType.ONLY_IN_GAME)
-@CommandParameters(description = "Pretty rainbow trails.", usage = "/<command> [off]")
+@CommandParameters(description = "Pretty rainbow trails.", usage = "/<command> [off | glass]")
 public class Command_trail extends TFM_Command
 {
-    private static Listener movementListener = null;
+    public static Listener movementListener = null;
     private static final List<Player> trailPlayers = new ArrayList<Player>();
     private static final Random RANDOM = new Random();
 
     @Override
     public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
     {
+        if (args.length > 0 && "glass".equals(args[0]))
+        {
+            Bukkit.getPluginManager().registerEvents(movementListener = new Listener()
```

Comment:
The implementation is a duplicate of the wool trail. The implementation should probably be in the same method, with a variable defining the base material, on which to apply the data byte.